### PR TITLE
kafka-util: don't report librdkafka errors to Sentry

### PR DESCRIPTION
**Pull request by benesch**
They are far too spammy, and generally indicate user configuration errors rather than actual bugs in Materialize that need to be reported on.

### Motivation

  * This PR fixes an issue reported on Slack: librdkafka errors are too spammy. See: https://materializeinc.slack.com/archives/C01CFKM1QRF/p1696256524616059?thread_ts=1696255775.857209&cid=C01CFKM1QRF

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - n/a

### CODE REVIEW

**Review by petrosagg — APPROVED** [review]
LGTM!

**Review by nrainer-materialize — APPROVED** [review]

File: src/kafka-util/src/client.rs [hunk 1]
```diff
@@ -160,9 +160,13 @@ impl ClientContext for MzClientContext {
         match level {
             Emerg | Alert | Critical | Error => {
                 self.record_error(log_message);
-                error!(target: "librdkafka", "{} {}", fac, log_message);
+                // We downgrade error messages to `warn!` level to avoid
+                // sending the errors to Sentry. Most errors are customer
+                // configuration problems that are not appropriate to send to
+                // Sentry.
+                warn!(target: "librdkafka", "error: {} {}", fac, log_message);
```
Comment (guswynn) [inline src/kafka-util/src/client.rs:167]: alternatively we can just add a per-layer filter to the sentry layer to filter librdkafka?

Comment (benesch) [reply src/kafka-util/src/client.rs:167]: I thought about that! But I like the rule we have currently where error! logs are 1:1 with what gets reported to Sentry. And our log level guidance says:

> `ERROR`: Reports an error that certainly indicates data corruption, data loss, unavailability, or an invariant violation that results in undefined behavior.

So by that guidance we shouldn't be using the `error!` log level, as these logs do not _certainly_ indicate any of the above.

Comment (guswynn) [reply src/kafka-util/src/client.rs:167]: ah yes, sounds good!

**Pull request merged**